page/upgrade-guide/1.12.md

```diff
@@ -0,0 +1,353 @@
+
+### Dropped support for jQuery 1.6
+
+[(#10723)](http://bugs.jqueryui.com/ticket/10723) jQuery UI no longer supports jQuery 1.6. The minimal supported version is now 1.7.0.
+
+## Core
+
+### Removed .focus( delay )
+
+[(#9649)](http://bugs.jqueryui.com/ticket/9649) The deprecated `.focus( delay )` method override has been removed. jQuery UI was using this only in our dialog widget, where we've replaced the delayed focus call with a timeout.
+
+### Removed .zIndex()
+
+[(#9156)](http://bugs.jqueryui.com/ticket/9156) The deprecated `.zIndex()` method has been removed, in favor of the new `.ui-front` / `appendTo` logic used by all widgets that display an element on top of the page, like dialog and autocomplete.
+
+This method was used by dialog, but because the dialog now has an [`appendTo` option](http://api.jqueryui.com/dialog/#option-appendTo), this plugin method is no longer necessary. If you are using `.zIndex()`, or building a widget that must stack, check out our [guide to create stacking elements with a `ui-front` class name and `appendTo` option](http://api.jqueryui.com/theming/stacking-elements/).
+
+### Added .labels()
+
+[(#12475)](http://bugs.jqueryui.com/ticket/12475) This release introduces a new jQuery plugin method. `.labels()` finds all label elements associated with the first selected element.
+
+## Mouse
+
+### Deprecate distance and delay options
+
+[(#10615)](http://bugs.jqueryui.com/ticket/10615) Interactions should be instantaneous. These settings are typically used to prevent accidental drags, but a proper fix for that is to improve the UX, e.g., using handles to avoid accidental drags.
+
+## Button
+
+The button widget underwent a rewrite in this release. Most notably, support for inputs of type checkbox and radio has been extracted into a standalone [checkboxradio widget](http://api.jqueryui.com/checkboxradio/), and support for groups of buttons has been extracted into a more generic [controlgroup widget](http://api.jqueryui.com/controlgroup/). The remaining button widget also underwent some changes, simplifiying markup and options.
+
+### Deprecated `icons` options in favor of `icon` and `iconPosition`
+
+The button widget used to have an `icons` option that supported `primary` and `secondary` properties, allowing up to two icons at once. This has been replaced by the `icon` option, which only allows a single icon, and the `iconPosition` option, which controls the placement of that one icon.
+
+The equivalent of the old `primary` icons property is `iconPosition: "beginning"`, while the equivalent of the old `secondary` icons property is `iconPosition: "end"`. The new button also supports "top" and "bottom" for `iconPosition`, allowing the icon to be displayed above or below the button text.
+
```

Comment:
we should note here that you can still have 2 icons like in 1.11 but you need to add them your self along with an example.